src/knn.jl

```diff
@@ -1,4 +1,11 @@
-check_k(tree, k) = (k > length(tree.data)|| k <= 0) && throw(ArgumentError("k > number of points in tree or ≦ 0"))
+function check_k(tree, k)
+    if isempty(tree.data)
+        k = 0
```

Comment:
This seems like a weird special case to me. Either k should be allowed to be bigger than the number of points and in that case we truncate it to be equal to the number of points or we error.